src/resolve.ts

```diff
@@ -21,7 +21,7 @@ function retrieveTypeName (
   type: Type
 ): string {
   if (type.isArray()) {
-    return retrieveTypeName(type.getArrayElementType()!)
+    return `Array_${retrieveTypeName(type.getArrayElementType()!)}`
```

Comment:
Using plural of the type name would probably be better

Reply:
Not sure how i could use the plural form of the type ? adding a `s` wont work for some types (like `Entity` should be  `Entities` and not `Entitys` )